p2p/src/authenticated/discovery/actors/listener.rs

```diff
@@ -94,20 +94,19 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Network + Rng + CryptoRng + Metri
                 return;
             }
         };
-        span.record("peer", incoming.peer().to_string());
-
-        // Attempt to claim the connection
-        //
-        // Reserve also checks if the peer is authorized.
         let peer = incoming.peer();
-        let Some(reservation) = tracker
-            .listen(peer.clone())
-            .instrument(debug_span!("reserve"))
+        span.record("peer", peer.to_string());
+        debug!(?peer, "verified handshake");
+
+        // Check if the peer is listenable
+        if !tracker
```

Comment:
There is extra overhead from messaging the tracker twice per handshake (rather than the once before).